airbrakes/imu/imu.py

```diff
@@ -0,0 +1,136 @@
+            for packet in packets:
+                # The data packet from the IMU:
+                packet: mscl.MipDataPacket
+
+                # Get the timestamp of the packet
+                timestamp = packet.collectedTimestamp().nanoseconds()
+
+                # Initialize packet with the timestamp, determines if the packet is raw or estimated
+                imu_data_packet = (
+                    EstimatedDataPacket(timestamp)
+                    if packet.descriptorSet() == self.ESTIMATED_DESCRIPTOR_SET
+                    else RawDataPacket(timestamp)
+                )
+
+                # Each of these packets has multiple data points
+                for data_point in packet.data():
+                    data_point: mscl.MipDataPoint
+                    if data_point.valid():
+                        channel = data_point.channelName()
+                        # This cpp file was the only place I was able to find all the channel names
+                        # https://github.com/LORD-MicroStrain/MSCL/blob/master/MSCL/source/mscl/MicroStrain/MIP/MipTypes.cpp
+                        # Check if the imu_data_packet has an attribute with the name of the channel
+                        if hasattr(imu_data_packet, channel):
+                            # First checks if the data point needs special handling, if not, just set the attribute
+                            match channel:
+                                # These specific data points are matrix's rather than doubles
+                                case "estAttitudeUncertQuaternion" | "estOrientQuaternion":
+                                    matrix = data_point.as_Matrix()
+                                    # Converts the [4x1] matrix to the X, Y, Z, and W of the quaternion
+                                    quaternion_tuple = tuple(matrix[i, 0] for i in range(matrix.rows()))
+                                    # Sets the X, Y, Z, and W of the quaternion to the data packet object
+                                    setattr(imu_data_packet, f"{channel}X", quaternion_tuple[0])
+                                    setattr(imu_data_packet, f"{channel}Y", quaternion_tuple[1])
+                                    setattr(imu_data_packet, f"{channel}Z", quaternion_tuple[2])
+                                    setattr(imu_data_packet, f"{channel}W", quaternion_tuple[3])
+                                case _:
+                                    # Because the attribute names in our data packet classes are the same as the channel
+                                    # names, we can just set the attribute to the value of the data point.
+                                    setattr(imu_data_packet, channel, data_point.as_float())
+
+                # Put the latest data into the shared queue
+                self.data_queue.put(imu_data_packet)
+
+    def get_imu_data_packet(self) -> IMUDataPacket | EstimatedDataPacket:
```

Comment:
Shouldn't this just return IMUDataPacket?

Reply:
oops yes.